File: hibernate-examples/src/test/java/org/hibernate/examples/AbstractJpaTest.java
/*
 * Copyright (c) 2017. Sunghyouk Bae <sunghyouk.bae@gmail.com>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package org.hibernate.examples;

import lombok.extern.slf4j.Slf4j;
import org.hibernate.examples.config.JpaH2Configuration;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringRunner;

import javax.persistence.EntityManagerFactory;

/**
 * org.hibernate.examples.AbstractJpaTest
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 2013. 11. 27. 오후 5:45
 */
@Slf4j
@RunWith(SpringRunner.class)
@ContextConfiguration(classes = {JpaH2Configuration.class})
public abstract class AbstractJpaTest {

  @Autowired
  protected EntityManagerFactory emf;
}

File: hibernate-scala-examples/src/test/scala/org/hibernate/examples/tests/AbstractJpaTest.scala
package org.hibernate.examples.tests

import org.junit.runner.RunWith
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner
import org.springframework.test.context.ContextConfiguration
import org.hibernate.examples.tests.config.JpaH2Configuration
import org.springframework.beans.factory.annotation.Autowired
import javax.persistence.EntityManagerFactory

/**
 * org.hibernate.examples.tests.AbstractJpaTest 
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 2014. 1. 9. 오후 4:12
 */
@RunWith(classOf[SpringJUnit4ClassRunner])
@ContextConfiguration(classes = Array(classOf[JpaH2Configuration]))
abstract class AbstractJpaTest {

    @Autowired
    val emf: EntityManagerFactory
}
